subscribers.js

```diff
@@ -73,6 +74,74 @@ const Subscribers = {
       return _ret(e);
     }
   },
+
+  async addTempUser(aEmail, aToken) {
+    if (!this.validateToken(aToken)) {
+      throw new Error("Invalid token, are you using Subscribers.generateToken?");
+    }
+
+    // Error code for attempting to add a duplicate entry on a UNIQUE key.
+    const DUPLICATE_ERROR = "23505";
+    try {
+      await dbq({
+        text: "INSERT INTO users_temp ( email, token ) VALUES ( $1, $2 );",
+        values: [ aEmail, aToken ],
+      });
+
+      return _ret();
+    } catch (e) {
+      if (e.code === DUPLICATE_ERROR) {
+        // Duplicate entry, count it as success.
+        return _ret(null, { duplicate: true });
+      }
+
+      return _ret(e);
+    }
+  },
+
+  async deleteTempUser(aEmail) {
+    try {
+      await dbq({
+        text: "DELETE FROM users_temp WHERE email = $1;",
+        values: [ aEmail ],
+      });
+      return _ret();
+    } catch(e) {
+      return _ret(e);
+    }
+  },
+
+  async getTempUser(aEmail) {
+    try {
+      const rows = (await dbq({
```

Comment:
Not sure if we could just destructure `const {rows}` here, and save some brackets and `.rows` property. Maybe.

```js
const {rows} = await dbq({
  text: "SELECT * FROM users_temp WHERE email = $1;",
  values: [ aEmail ],
});
```

Reply:
Nice

Reply:
Migrated comment to https://github.com/mozilla/blurts-server/pull/64/files#r173320471